interpreter/src/main/scala/pl/touk/nussknacker/engine/definition/TypeInfos.scala

```diff
@@ -1,103 +1,85 @@
 package pl.touk.nussknacker.engine.definition
 
-import cats.data.ValidatedNel
+import cats.data.{NonEmptyList, ValidatedNel}
 import cats.implicits.catsSyntaxValidatedId
-import io.circe.Encoder
-import io.circe.generic.JsonCodec
-import pl.touk.nussknacker.engine.api.generics.{ExpressionParseError, GenericFunctionTypingError, Parameter, ParameterList}
-import pl.touk.nussknacker.engine.api.typed.TypeEncoders
+import pl.touk.nussknacker.engine.api.generics.GenericFunctionTypingError.ArgumentTypeError
+import pl.touk.nussknacker.engine.api.generics.{ExpressionParseError, GenericFunctionTypingError, MethodTypeInfo, Parameter}
 import pl.touk.nussknacker.engine.api.typed.typing.{Typed, TypedClass, TypingResult}
 import pl.touk.nussknacker.engine.spel.SpelExpressionParseErrorConverter
 
 object TypeInfos {
-  //a bit sad that it isn't derived automatically, but...
-  private implicit val tce: Encoder[TypedClass] = TypeEncoders.typingResultEncoder.contramap[TypedClass](identity)
-
-  @JsonCodec(encodeOnly = true) case class SerializableMethodInfo(parameters: List[Parameter],
-                                                                  refClazz: TypingResult,
-                                                                  description: Option[String],
-                                                                  varArgs: Boolean)
-
-  implicit val methodInfoEncoder: Encoder[MethodInfo] = Encoder[SerializableMethodInfo].contramap(_.serializable)
-
   sealed trait MethodInfo {
     def computeResultType(arguments: List[TypingResult]): ValidatedNel[ExpressionParseError, TypingResult]
 
-    def staticParameters: ParameterList
-
-    final def staticNoVarArgParameters: List[Parameter] = staticParameters.noVarArgs
-
-    final def staticVarArgParameter: Option[Parameter] = staticParameters.varArg
+    def signatures: NonEmptyList[MethodTypeInfo]
 
-    def staticResult: TypingResult
+    val mainSignature: MethodTypeInfo = signatures.toList.maxBy(_.parametersToList.length)
```

Comment:
This shouldn't be computed on this level. For StaticMethodInfo it is just signature, no computation is needed + add comment / rename as we discussed